Thanks - I think it must be a different issue here, because both the JAX and scipy results are float arrays, but the values do not match. e.g.
```
 (shape=(10, 5, 21), axis=2, dtype=<class 'numpy.float32'>, contains_nans=True, method='max')
```
```
AssertionError: 
Not equal to tolerance rtol=5e-06, atol=5e-06

Mismatched elements: 56 / 1050 (5.33%)
Max absolute difference: 5
Max relative difference: 0.3125
 x: array([[[10, 16, 14, ...,  7, 17,  4],
        [ 9, 21, 10, ...,  4,  6, 18],
        [ 5, 20, 10, ..., 21,  8, 15],...
 y: array([[[10, 16, 14, ...,  7, 17,  4],
        [ 9, 20, 10, ...,  4,  6, 18],
        [ 5, 17, 10, ..., 21,  8, 15],...
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,9 @@ Remember to align the itemized text with the first line of an item within a list
     * Deleted {func}`jax.scipy.linalg.polar_unitary`, which was a deprecated JAX 
       extension to the scipy API. Use {func}`jax.scipy.linalg.polar` instead.
 
+  * Changes
+    * Added {func}`jax.scipy.stats.rankdata`.
+
 ## jaxlib 0.4.3
 
 ## jax 0.4.2 (Jan 24, 2023)

--- a/docs/jax.scipy.rst
+++ b/docs/jax.scipy.rst
@@ -148,6 +148,7 @@ jax.scipy.stats
   :toctree: _autosummary
 
    mode
+   rankdata
 
 jax.scipy.stats.bernoulli
 ~~~~~~~~~~~~~~~~~~~~~~~~~

--- a/jax/_src/scipy/stats/_core.py
+++ b/jax/_src/scipy/stats/_core.py
@@ -1,4 +1,4 @@
-# Copyright 2022 The JAX Authors.
+# Copyright 2023 The JAX Authors.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -23,7 +23,7 @@ from jax._src import dtypes
 from jax._src.api import vmap
 from jax._src.numpy.lax_numpy import _check_arraylike
 from jax._src.numpy.util import _wraps
-from jax._src.typing import ArrayLike
+from jax._src.typing import ArrayLike, Array
 from jax._src.util import canonicalize_axis, prod
 
 ModeResult = namedtuple('ModeResult', ('mode', 'count'))
@@ -39,7 +39,7 @@ def mode(a: ArrayLike, axis: Optional[int] = 0, nan_policy: str = "propagate", k
   if nan_policy not in ["propagate", "omit", "raise"]:
     raise ValueError(
       f"Illegal nan_policy value {nan_policy!r}; expected one of "
-      "{'propoagate', 'omit', 'raise'}"
+      "{'propagate', 'omit', 'raise'}"
     )
   if nan_policy == "omit":
     # TODO: return answer without nans included.
@@ -81,3 +81,65 @@ def mode(a: ArrayLike, axis: Optional[int] = 0, nan_policy: str = "propagate", k
   x = x.reshape(x.shape[0], prod(x.shape[1:]))
   vals, counts = vmap(_mode_helper, in_axes=1)(x)
   return ModeResult(vals.reshape(output_shape), counts.reshape(output_shape))
+
+def invert_permutation(i: Array) -> Array:
+  """Helper function that inverts a permutation array."""
+  return jnp.empty_like(i).at[i].set(jnp.arange(i.size, dtype=i.dtype))
+
+@_wraps(scipy.stats.rankdata, lax_description="""\
+Currently the only supported nan_policy is 'propagate'
+""")
+@partial(jit, static_argnames=["method", "axis", "nan_policy"])
+def rankdata(
+  a: ArrayLike,
+  method: str = "average",
+  *,
+  axis: Optional[int] = None,
+  nan_policy: str = "propagate",
+) -> Array:
+
+  _check_arraylike("rankdata", a)
+
+  if nan_policy not in ["propagate", "omit", "raise"]:
+    raise ValueError(
+      f"Illegal nan_policy value {nan_policy!r}; expected one of "
+      "{'propoagate', 'omit', 'raise'}"
+    )
+  if nan_policy == "omit":
+    raise NotImplementedError(
+      f"Logic for `nan_policy` of {nan_policy} is not implemented"
+    )
+  if nan_policy == "raise":
+    raise NotImplementedError(
+      "In order to best JIT compile `mode`, we cannot know whether `x` "
+      "contains nans. Please check if nans exist in `x` outside of the "
+      "`rankdata` function."
+    )
+
+  if method not in ("average", "min", "max", "dense", "ordinal"):
+    raise ValueError(f"unknown method '{method}'")
+
+  a = jnp.asarray(a)
+
+  if axis is not None:
+    return jnp.apply_along_axis(rankdata, axis, a, method)
+
+  arr = jnp.ravel(a)
+  sorter = jnp.argsort(arr)
+  inv = invert_permutation(sorter)
+
+  if method == "ordinal":
+    return inv + 1
+  arr = arr[sorter]
+  obs = jnp.insert(arr[1:] != arr[:-1], 0, True)
+  dense = obs.cumsum()[inv]
+  if method == "dense":
+    return dense
+  count = jnp.nonzero(obs, size=arr.size + 1, fill_value=len(obs))[0]
+  if method == "max":
+    return count[dense]
+  if method == "min":
+    return count[dense - 1] + 1
+  if method == "average":
+    return .5 * (count[dense] + count[dense - 1] + 1).astype(dtypes.canonicalize_dtype(jnp.float_))
+  raise ValueError(f"unknown method '{method}'")

--- a/jax/scipy/stats/__init__.py
+++ b/jax/scipy/stats/__init__.py
@@ -37,5 +37,5 @@ from jax.scipy.stats import betabinom as betabinom
 from jax.scipy.stats import gennorm as gennorm
 from jax.scipy.stats import truncnorm as truncnorm
 from jax._src.scipy.stats.kde import gaussian_kde as gaussian_kde
-from jax._src.scipy.stats._core import mode as mode
+from jax._src.scipy.stats._core import mode as mode, rankdata as rankdata
 from jax.scipy.stats import vonmises as vonmises

--- a/tests/scipy_stats_test.py
+++ b/tests/scipy_stats_test.py
@@ -1129,5 +1129,26 @@ class LaxBackedScipyStatsTests(jtu.JaxTestCase):
                             tol=tol)
     self._CompileAndCheck(lax_fun, args_maker, rtol=tol)
 
+  @jtu.sample_product(
+    [dict(shape=shape, axis=axis)
+      for shape in [(0,), (7,), (47, 8), (0, 2, 3), (10, 5, 21)]
+      for axis in [None, *range(len(shape))
+    ]],
+    dtype=jtu.dtypes.integer + jtu.dtypes.floating,
+    method=['average', 'min', 'max', 'dense', 'ordinal']
+  )
+  def testRankData(self, shape, dtype, axis, method):
+
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: [rng(shape, dtype)]
+
+    scipy_fun = partial(osp_stats.rankdata, method=method, axis=axis)
+    lax_fun = partial(lsp_stats.rankdata, method=method, axis=axis)
+    tol_spec = {np.float32: 2e-4, np.float64: 5e-6}
+    tol = jtu.tolerance(dtype, tol_spec)
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker, check_dtypes=False,
+                            tol=tol)
+    self._CompileAndCheck(lax_fun, args_maker, rtol=tol)
+
 if __name__ == "__main__":
   absltest.main(testLoader=jtu.JaxTestLoader())

